test-tests/shared/src/test/scala/zio/test/BoolAlgebraSpec.scala

```diff
@@ -0,0 +1,167 @@
+package zio.test
+
+import zio.test.Assertion._
+import zio.test.BoolAlgebraSpecHelper._
+import zio.test.TestAspect._
+
+import scala.util.Random
+
+object BoolAlgebraSpec
```

Comment:
Now that we are using `ZIO Test` we can take advantage of its built in property based testing functionality instead of our homegrown one. So `randomBoolAlgebra` can return a `Gen[Random, BoolAlgebra[Int]]` and then in all the tests that use it we can use the `check` method instead of `nonFlaky`.

Reply:
great idea!